### PR TITLE
Fix: check space after yield keyword in space-unary-ops (fixes #2707)

### LINKED ISSUE #2707: space-unary-ops should enforce yield spacing
I would expect `space-unary-ops` (with the default options) to report `yield(0)` and `yield+0`. I would probably expect it to report `yield*0` and the like as well.

### PR DESCRIPTION
**Pull request by lo1tuma**

### CODE REVIEW

File: lib/rules/space-unary-ops.js [hunk 1]
```diff
@@ -65,7 +65,8 @@ module.exports = function(context) {
             firstToken = tokens[0],
             secondToken = tokens[1];
 
-        if ((node.type === "NewExpression" || node.prefix) && firstToken.type === "Keyword") {
+
+        if (((node.type === "NewExpression" || node.prefix) && firstToken.type === "Keyword") || (firstToken.type === "Identifier" && firstToken.value === "yield")) {
```
Comment (lo1tuma) [inline lib/rules/space-unary-ops.js]: I think there is a bug in espree, the token for the `yield` keyword shouldn’t be `Identifier`. I’ve filed an espree issue eslint/espree#190

File: tests/lib/rules/space-unary-ops.js [hunk 2]
```diff
@@ -123,6 +123,18 @@ ruleTester.run("space-unary-ops", rule, {
...
+        {
+            code: "function *foo () { yield (0) }",
+            ecmaFeatures: { generators: true }
+        },
+        {
+            code: "function *foo() { yield +1 }",
+            ecmaFeatures: { generators: true }
+        },
+        {
+            code: "function *foo() { yield *0 }",
```
Comment (michaelficarra) [inline tests/lib/rules/space-unary-ops.js]: Space here should be after `*`.

Comment (lo1tuma) [inline tests/lib/rules/space-unary-ops.js]: Just to be sure, the following examples should be valid:

```js
function *foo() { yield* 0 }
function *foo() { yield * 0 }
```

and those invalid:

```js
function *foo() { yield*0 }
function *foo() { yield *0 }
```

:question:

Comment (michaelficarra) [inline tests/lib/rules/space-unary-ops.js]: Yeah.

**Comment by nzakas** [comment]
@lo1tuma still working on this?

**Comment by lo1tuma** [comment]
@nzakas Sure, I will continue today.

**Comment by lo1tuma** [comment]
I’ve updated the PR.

**Comment by michaelficarra** [comment]
LGTM.

**Pull request merged**